apps/emqx_utils/src/emqx_utils_stream.erl

```diff
@@ -91,6 +93,31 @@ map(F, S) ->
         end
     end.
 
+%% @doc Zip a list of streams into a stream producing lists of their respective values.
+%% The resulting stream is as long as the shortest of the input streams.
+-spec zip([stream(X)]) -> stream([X]).
```

Comment:
Nit: with this name, I'd expect the typespec to be something like `zip([stream(_)]) -> stream(tuple())`, similar to `lists:zip`, `Enum.zip` and others.  But I see how this has the same "shape" as a stream of tuples.  :thinking: 

Also, this function seems to be unused in this PR?

Reply:
I should probably rename it. Agree that it brokes expectations of what _zip_s are, however I found that having it emit rows as lists is a bit more usable overall, especially given it's generic in the number of things being zipped (e.g. one can do `zipwith/2` through `erlang:apply/2`).

> Also, this function seems to be unused in this PR?

Yeah it was used in the previous iteration. Do you think it's better to drop it?